Kicked off DataStax CI build to test and I'm getting build errors from the test:

```

[2025-01-16T21:48:26.283Z] [INFO] -------------------------------------------------------------
[2025-01-16T21:48:26.283Z] [INFO] -------------------------------------------------------------
[2025-01-16T21:48:26.283Z] [ERROR] COMPILATION ERROR : 
[2025-01-16T21:48:26.283Z] [INFO] -------------------------------------------------------------
[2025-01-16T21:48:26.283Z] [ERROR] /home/jenkins/workspace/drivers_java_oss_PR-1981/driver-core/src/test/java/com/datastax/driver/core/ControlConnectionTest.java:[695,22] cannot find symbol
[2025-01-16T21:48:26.283Z]   symbol:   method getManager()
[2025-01-16T21:48:26.283Z]   location: variable cluster of type com.datastax.driver.core.Cluster
[2025-01-16T21:48:26.283Z] [INFO] 1 error
```

Haven't been able to repro locally as this branch is still using the old tcnative impl and I'm on an ARM processor.  I'll see if I can work around that to try and repro (and fix) locally.

--- a/driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java
+++ b/driver-core/src/main/java/com/datastax/driver/core/ControlConnection.java
@@ -626,7 +626,11 @@ class ControlConnection implements Connection.Owner {
       broadcastRpcAddress = new InetSocketAddress(nativeAddress, nativePort);
     } else if (row.getColumnDefinitions().contains("rpc_address")) {
       InetAddress rpcAddress = row.getInet("rpc_address");
-      broadcastRpcAddress = new InetSocketAddress(rpcAddress, cluster.connectionFactory.getPort());
+      int nativePort = cluster.connectionFactory.getPort();
+      if (row.getColumnDefinitions().contains("rpc_port")) {
+        nativePort = row.getInt("rpc_port");
+      }
+      broadcastRpcAddress = new InetSocketAddress(rpcAddress, nativePort);
     }
     // Before CASSANDRA-9436, system.local doesn't have rpc_address, so this might be null. It's not
     // a big deal because we only use this for server events, and the control node doesn't receive
@@ -854,8 +858,11 @@ class ControlConnection implements Connection.Owner {
         broadcastRpcAddress = new InetSocketAddress(nativeAddress, nativePort);
       } else {
         InetAddress rpcAddress = row.getInet("rpc_address");
-        broadcastRpcAddress =
-            new InetSocketAddress(rpcAddress, cluster.connectionFactory.getPort());
+        int nativePort = cluster.connectionFactory.getPort();
+        if (row.getColumnDefinitions().contains("rpc_port")) {
+          nativePort = row.getInt("rpc_port");
+        }
+        broadcastRpcAddress = new InetSocketAddress(rpcAddress, nativePort);
       }
       broadcastRpcAddresses.add(broadcastRpcAddress);
 

--- a/driver-core/src/test/java/com/datastax/driver/core/ControlConnectionTest.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/ControlConnectionTest.java
@@ -20,6 +20,7 @@ package com.datastax.driver.core;
 import static com.datastax.driver.core.Assertions.assertThat;
 import static com.datastax.driver.core.CreateCCM.TestMode.PER_METHOD;
 import static com.datastax.driver.core.ScassandraCluster.SELECT_LOCAL;
+import static com.datastax.driver.core.ScassandraCluster.SELECT_LOCAL_RPC_ADDRESS_AND_PORT;
 import static com.datastax.driver.core.ScassandraCluster.SELECT_PEERS;
 import static com.datastax.driver.core.ScassandraCluster.SELECT_PEERS_DSE68;
 import static com.datastax.driver.core.ScassandraCluster.SELECT_PEERS_V2;
@@ -657,6 +658,50 @@ public class ControlConnectionTest extends CCMTestsSupport {
             .expectedAddress(expectedAddress)
             .build();
     runPeerTest(state);
+  }
+
+  @Test(groups = "short")
+  @CCMConfig(createCcm = false)
+  public void should_extract_hosts_port_using_rpc_port_from_local() throws UnknownHostException {
+    InetAddress expectedAddress = InetAddress.getByName("1.2.3.4");
+    int expectedPort = 29042;
+    PeerRowState state =
+        PeerRowState.builder()
+            .local("rpc_address", expectedAddress)
+            .local("rpc_port", expectedPort)
+            .build();
+
+    ScassandraCluster scassandras =
+        ScassandraCluster.builder().withNodes(2).withPeersV2(state.usePeersV2()).build();
+    scassandras.init();
+
+    Cluster cluster = null;
+    try {
+      scassandras.node(1).primingClient().clearAllPrimes();
+      PrimingClient primingClient = scassandras.node(1).primingClient();
+      primingClient.prime(
+          PrimingRequest.queryBuilder()
+              .withQuery("SELECT * FROM system.local WHERE key='local'")
+              .withThen(
+                  then()
+                      .withColumnTypes(SELECT_LOCAL_RPC_ADDRESS_AND_PORT)
+                      .withRows(state.getLocalRow())
+                      .build())
+              .build());
+      cluster =
+          Cluster.builder()
+              .addContactPoints(scassandras.address(1).getAddress())
+              .withPort(scassandras.getBinaryPort())
+              .withNettyOptions(nonQuietClusterCloseOptions)
+              .build();
+      cluster.connect();
+
+      assertThat(cluster.manager.getControlConnection().connectedHost().getBroadcastRpcAddress())
+          .isEqualTo(new InetSocketAddress(expectedAddress, expectedPort));
+    } finally {
+      if (cluster != null) cluster.close();
+      scassandras.stop();
+    }
   }
 
   private void runPeerTest(PeerRowState state) {

--- a/driver-core/src/test/java/com/datastax/driver/core/ScassandraCluster.java
+++ b/driver-core/src/test/java/com/datastax/driver/core/ScassandraCluster.java
@@ -658,14 +658,14 @@ public class ScassandraCluster {
     column("cluster_name", TEXT),
     column("cql_version", TEXT),
     column("data_center", TEXT),
+    column("graph", BOOLEAN),
+    column("host_id", UUID),
     column("listen_address", INET),
     column("partitioner", TEXT),
     column("rack", TEXT),
     column("release_version", TEXT),
-    column("tokens", set(TEXT)),
-    column("graph", BOOLEAN),
-    column("host_id", UUID),
-    column("schema_version", UUID)
+    column("schema_version", UUID),
+    column("tokens", set(TEXT))
   };
 
   public static final org.scassandra.http.client.types.ColumnMetadata[] SELECT_LOCAL_V2 = {
@@ -676,15 +676,36 @@ public class ScassandraCluster {
     column("cluster_name", TEXT),
     column("cql_version", TEXT),
     column("data_center", TEXT),
+    column("graph", BOOLEAN),
+    column("host_id", UUID),
     column("listen_address", INET),
     column("listen_port", INT),
     column("partitioner", TEXT),
     column("rack", TEXT),
     column("release_version", TEXT),
-    column("tokens", set(TEXT)),
-    column("graph", BOOLEAN),
+    column("schema_version", UUID),
+    column("tokens", set(TEXT))
+  };
+
+  public static final org.scassandra.http.client.types.ColumnMetadata[]
+      SELECT_LOCAL_RPC_ADDRESS_AND_PORT = {
+    column("key", TEXT),
+    column("bootstrapped", TEXT),
+    column("broadcast_address", INET),
+    column("broadcast_port", INT),
+    column("cluster_name", TEXT),
+    column("cql_version", TEXT),
+    column("data_center", TEXT),
     column("host_id", UUID),
-    column("schema_version", UUID)
+    column("listen_address", INET),
+    column("listen_port", INT),
+    column("partitioner", TEXT),
+    column("rack", TEXT),
+    column("release_version", TEXT),
+    column("rpc_address", INET),
+    column("rpc_port", INT),
+    column("schema_version", UUID),
+    column("tokens", set(TEXT))
   };
 
   static final org.scassandra.http.client.types.ColumnMetadata[] SELECT_CLUSTER_NAME = {

